sql/src/main/java/io/crate/planner/optimizer/Optimizer.java

```diff
@@ -52,9 +56,15 @@ private LogicalPlan tryApplyRules(LogicalPlan plan) {
             for (Rule rule : rules) {
                 Match match = rule.pattern().accept(node, Captures.empty());
                 if (match.isPresent()) {
+                    if (LOGGER.isTraceEnabled()) {
```

Comment:
Maybe store the result once before the `while` to avoid the repeated checks. We don't have to account for config changes within a `tryApplyRules` call